flink-runtime/src/main/java/org/apache/flink/runtime/state/TaskLocalStateStore.java

```diff
@@ -19,92 +19,224 @@
 package org.apache.flink.runtime.state;
 
 import org.apache.flink.api.common.JobID;
-import org.apache.flink.runtime.checkpoint.CheckpointMetaData;
 import org.apache.flink.runtime.checkpoint.TaskStateSnapshot;
 import org.apache.flink.runtime.jobgraph.JobVertexID;
-import org.apache.flink.util.ExceptionUtils;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnegative;
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
+import javax.annotation.concurrent.GuardedBy;
 
 import java.io.File;
-import java.util.HashMap;
+import java.util.Arrays;
+import java.util.Iterator;
 import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executor;
 
 /**
  * This class will service as a task-manager-level local storage for local checkpointed state. The purpose is to provide
  * access to a state that is stored locally for a faster recovery compared to the state that is stored remotely in a
  * stable store DFS. For now, this storage is only complementary to the stable storage and local state is typically
  * lost in case of machine failures. In such cases (and others), client code of this class must fall back to using the
  * slower but highly available store.
- *
- * TODO this is currently a placeholder / mock that still must be implemented!
  */
 public class TaskLocalStateStore {
 
-	/** */
+	/** Logger for this class. */
+	private static final Logger LOG = LoggerFactory.getLogger(TaskLocalStateStore.class);
+
+	/** Maximum number of retained snapshots. */
+	private static final int MAX_RETAINED_SNAPSHOTS = 5;
```

Comment:
How did you come up with this number? Should this be configurable?

Reply:
This can be removed without replacement.